lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1703,12 +1813,25 @@ LogicalResult ReduceOpConversion::matchAndRewriteBasic(
           select(readMask, linearize(rewriter, loc, readIdx, smemShape, srcOrd),
                  ints[0]);
       Value readPtr = gep(elemPtrTy, writePtr, readOffset);
+      Value indexReadPtr = gep(indexPtrTy, indexWritePtr, readOffset);
```

Comment:
Nit. the definition of `IndexReadPtr` actually can be put into the else block.

Reply:
Done